packages/wdio-junit-reporter/tests/reporter.test.js

```diff
@@ -72,21 +84,56 @@ describe('wdio-junit-reporter', () => {
         reporter.suites = suitesLog
 
         // verifies the content of the report but omits format by stripping all whitespace and new lines
-        expect(reporter.prepareXml(runnerLog).replace(/\s/g, '')).toBe(testLog.toString().replace(/\s/g, ''))
+        expect(reporter.buildJunitXml(mochaRunnerLog).replace(/\s/g, '')).toMatchSnapshot(testLog.toString().replace(/\s/g, ''))
+    })
+
+    it('generates xml output (Cucumber-style)', () => {
+        reporter.suites = featuresLog
+
+        // verifies the content of the report but omits format by stripping all whitespace and new lines
+        expect(reporter.buildJunitXml(cucumberRunnerLog).replace(/\s/g, '')).toMatchSnapshot(testCucumberLog.toString().replace(/\s/g, ''))
+    })
+
+    it('scenario will be marked failed if a single scenario step fails (Cucumber-style)', () => {
+        reporter.suites = featuresWithFailingThenSkipStepLog
+
+        // verifies the content of the report but omits format by stripping all whitespace and new lines
+        expect(reporter.buildJunitXml(cucumberRunnerLog).replace(/\s/g, '')).toMatchSnapshot(testCucumberFailingThenSkippedStep.toString().replace(/\s/g, ''))
+    })
+
+    it('scenario will be marked failed if a single scenario step throws an error but there is no error object in JSON', () => {
+        reporter.suites = suitesWithNoErrorObjectLog
+
+        // verifies the content of the report but omits format by stripping all whitespace and new lines
+        expect(reporter.buildJunitXml(mochaRunnerLog).replace(/\s/g, '')).toMatchSnapshot(testWithNoErrorObjectLog.toString().replace(/\s/g, ''))
+    })
+
+    it('scenario will be marked failed if a single scenario step throws an error but there is no error object in JSON (Cucumber-style)', () => {
+        reporter.suites = featuresWithErrorStepAndNoErrorObjectLog
+
+        // verifies the content of the report but omits format by stripping all whitespace and new lines
+        expect(reporter.buildJunitXml(cucumberRunnerLog).replace(/\s/g, '')).toMatchSnapshot(testCucumberWithErrorStepAndNoErrorObject.toString().replace(/\s/g, ''))
+    })
+
+    it('scenario will be marked skipped if a single scenario step is pending (Cucumber-style)', () => {
+        reporter.suites = featuresWithPendingStepLog
+
+        // verifies the content of the report but omits format by stripping all whitespace and new lines
+        expect(reporter.buildJunitXml(cucumberRunnerLog).replace(/\s/g, '')).toMatchSnapshot(testCucumberWithPendingStep.toString().replace(/\s/g, ''))
```

Comment:
it looks like it makes sense to use `toMatchSnapshot` here, however it doesn't take any arguments so we can remove the fixtures.